templates/flutter-dio/services/service.dart.hbs

```diff
@@ -15,17 +15,19 @@ class {{title}}Service {
   {{title}}Service(this.client);
 
   Future<ApiPlatformPagedCollection<T>> getCollection<T extends {{title}}>({
-    Map<String, dynamic> queryParameters,
-    Options options,
-    CancelToken cancelToken,
-    ProgressCallback onReceiveProgress,
+    Map<String, dynamic>? queryParameters,
+    Options? options,
+    CancelToken? cancelToken,
+    ProgressCallback? onReceiveProgress,
   }) {
     if (options == null) {
       options = Options();
     }
     options.responseType = ResponseType.json;
     options.contentType = ContentType.json.toString();
-    options.headers['Accept'] = 'application/ld+json';
+    options.headers = {
+      'Accept': 'application/ld+json'
+    };
```

Comment:
This doesn't work because it's overwriting the headers provided externalli in case an options object is passed. You need to make sure `headers` is always initialized when creating an empty `Options` object and then overwrite only the Accept part.